modules/e2ee/Context.js

```diff
@@ -253,11 +249,9 @@ export class Context {
             newUint8.set(new Uint8Array(plainText), frameHeader.byteLength);
 
             encodedFrame.data = newData;
-        } catch (error) {
-            if (this._sharedKey) {
```

Comment:
Why was this check for shared key performed here?

Reply:
I think cause the rotation should be handled externally in sharedKey mode.

Reply:
Then shouldn't we just return in case of a shared key? We don't want to ratchet in that case.

Reply:
Yeah, true that. We should not return the encoded frame, but still return from the method.